svm/src/account_loader.rs

```diff
@@ -35,7 +35,13 @@ pub(crate) type TransactionRent = u64;
 pub(crate) type TransactionProgramIndices = Vec<Vec<IndexOfAccount>>;
 pub type TransactionCheckResult = Result<CheckedTransactionDetails>;
 pub type TransactionValidationResult = Result<ValidatedTransactionDetails>;
-pub type TransactionLoadResult = Result<LoadedTransaction>;
+
+#[derive(PartialEq, Eq, Debug)]
+pub enum TransactionLoadResult {
+    Loaded(LoadedTransaction),
+    FeesOnly(FeesOnlyTransaction),
+    NotLoaded(TransactionError),
+}
```

Comment:
Let me know what you think of the description I added. I think you're right that documenting how nonces are handled for fees-only transactions is going to be really important to avoid confusion, probably during the PR that adds support for committing such transactions

Reply:
Nvm, I just added a bit more to the `FeesOnly` comment to talk about nonces as well.